src/erl_optics.erl

```diff
@@ -33,12 +41,45 @@ counter_inc(Key, Amt) ->
     erl_optics_nif:counter_inc(Ptr, Amt).
 
 
+-spec counter_inc_alloc(binary()) -> ok | {error, term()}.
+
+counter_inc_alloc(Key) ->
+    counter_inc_alloc(Key, 1).
+
+-spec counter_inc_alloc(binary(), integer()) -> ok | {error, term()}.
+
+counter_inc_alloc(Key, Amt)->
+    {ok, OpticsPtr} = get_optics(),
+    {ok, Ptr} = erl_optics_nif:counter_alloc_get(OpticsPtr, Key),
+    erl_optics_nif:counter_inc(Ptr, Amt),
+    erl_optics_nif:lens_close(Ptr).
+
+
 -spec dist_record(binary(), float()) -> ok | {error, term()}.
 
 dist_record(Key, Val) ->
     {ok, Ptr} = get_lens(Key),
     erl_optics_nif:dist_record(Ptr, Val).
 
+-spec dist_record_alloc(binary(), float()) -> ok | {error, term()}.
+
+dist_record_alloc(Key, Val) ->
+    {ok, OpticsPtr} = get_optics(),
+    {ok, Ptr} = erl_optics_nif:dist_alloc_get(OpticsPtr, Key),
+    erl_optics_nif:dist_record(Ptr, Val),
+    erl_optics_nif:lens_close(Ptr).
+
+-spec dist_record_timing_now_us(binary(), float()) -> ok | {error, term()}.
```

Comment:
`os:timestamp()` produces `erlang:timesamp()`, not `float()`. I don't think this type spec is correct.